virtcontainers/device/drivers/block.go

```diff
@@ -169,5 +169,26 @@ func (device *BlockDevice) Dump() persistapi.DeviceState {
 	return ds
 }
 
+// Load loads DeviceState and converts it to specific device
+func (device *BlockDevice) Load(ds persistapi.DeviceState) {
+	device.GenericDevice = &GenericDevice{}
+	device.GenericDevice.Load(ds)
+
+	bd := ds.BlockDrive
+	if bd != nil {
```

Comment:
Nit: I tend to invert logic like this to reduce the indent and make the code as simple as possible:

```go

if bd == nil {
    return
}

device.BlockDrive = &config.BlockDrive{
      :
```

Same comment for some of the checks below...

Reply:
Oh, nice suggestion!